boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -196,6 +230,41 @@ class BmsStateMachineTest : public BaseStateMachineTest
         };
     }
 
+    void CheckInRangeCanSignalsInGivenState(
+        const State *state,
+        float        min_value,
+        float        max_value,
+        float &      fake_value,
+        float (*value_can_signal_getter)(const struct BmsCanTxInterface *),
+        uint8_t (*out_of_range_can_signal_getter)(
+            const struct BmsCanTxInterface *),
+        uint8_t ok_choice,
+        uint8_t underflow_choice,
+        uint8_t overflow_choice)
+    {
+        SetInitialState(state);
+
+        // Normal range
+        fake_value = (min_value + max_value) / 2;
+        LetTimePass(state_machine, 1000);
+        ASSERT_EQ(fake_value, value_can_signal_getter(can_tx_interface));
+        ASSERT_EQ(ok_choice, out_of_range_can_signal_getter(can_tx_interface));
+
+        // Underflow range
+        fake_value =
+            std::nextafter(min_value, std::numeric_limits<float>::lowest());
+        LetTimePass(state_machine, 1000);
+        ASSERT_EQ(
```

Comment:
What value is sent over CAN in the case of overflow/underflow?

Reply:
I'm missing the assertions. Good catch thanks

Reply:
Ok. I'm ready to approve after this then

Reply:
Actually this implementation is correct. Whats happening here is that fake_value is the return_val of one of the fake functions.

Reply:
It will never be larger or smaller than the min and max values specified in the dbc

Reply:
![image](https://user-images.githubusercontent.com/32230980/101438392-1f514800-38c7-11eb-904f-a04f9f130a1a.png)
context

Reply:
Yeah this is what I was wondering about with my initial comment. Wasn't sure how overflow/underflow cases were dealt with when sending CAN messages